core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigrationActor.scala

```diff
@@ -0,0 +1,857 @@
+    )
+
+  def retryFailuresLike(update: (DataAccess, Long) => ReadWriteAction[Any],
+                        failureMessage: String,
+                        others: String*
+  ): MigrateAction[Unit] =
+    for {
+      (maxAttempts, maxRetries) <- asks(d => (d.maxConcurrentAttempts, d.maxRetries))
+      now <- nowTimestamp
+      (migrationId, workspaceName, retryCount) <- inTransactionT { dataAccess =>
+        import dataAccess.{multiregionalBucketMigrationRetryQuery, multiregionalBucketMigrationQuery}
+        for {
+          (migrationId, workspaceName) <- multiregionalBucketMigrationRetryQuery.nextFailureLike(
+            maxRetries,
+            failureMessage,
+            others: _*
+          )
+          numAttempts <- OptionT.liftF(multiregionalBucketMigrationQuery.getNumActiveResourceLimitedMigrations)
+          if numAttempts < maxAttempts
+          retryCount <- OptionT.liftF[ReadWriteAction, Long] {
+            for {
+              MultiregionalBucketMigrationRetry(id, _, numRetries) <- multiregionalBucketMigrationRetryQuery
+                .getOrCreate(migrationId)
+              retryCount = numRetries + 1
+              _ <- multiregionalBucketMigrationRetryQuery.update(id,
+                                                                 multiregionalBucketMigrationRetryQuery.retriesCol,
+                                                                 retryCount
+              )
+              _ <- update(dataAccess, migrationId)
+            } yield retryCount
+          }
+        } yield (migrationId, workspaceName, retryCount)
+      }
+      _ <- getLogger[MigrateAction].info(
+        stringify(
+          "migrationId" -> migrationId,
+          "workspace" -> workspaceName,
+          "retry" -> retryCount,
+          "retried" -> now
+        )
+      )
+    } yield ()
+
+  // Create bucket in us-central1-a
+  final def createBucketInTargetRegion(migration: MultiregionalBucketMigration,
+                                       workspace: Workspace,
+                                       sourceBucketName: GcsBucketName,
+                                       destBucketName: GcsBucketName
+  ): MigrateAction[Unit] =
+    MigrateAction { env =>
+      def pollForBucketToBeCreated(interval: FiniteDuration, deadline: Deadline): IO[Unit] =
+        IO.sleep(interval).whileM_ {
+          for {
+            _ <- IO.raiseWhen(deadline.isOverdue()) {
+              WorkspaceMigrationException(
+                message = "Workspace migration failed: timed out waiting for bucket creation",
+                data = Map(
+                  "migrationId" -> migration.id,
+                  "workspace" -> workspace.toWorkspaceName,
+                  "googleProject" -> workspace.googleProjectId,
+                  "bucketName" -> destBucketName
+                )
+              )
+            }
+
+            // Don't need a requester pays project for the bucket in the new google project
```

Comment:
The bit referenced here is forthcoming in a future PR, correct?

Reply:
Yes, requester pays will be enabled on the bucket as part of [WOR-1098](https://broadworkbench.atlassian.net/browse/WOR-1098).

This comment also needs to be updated as the bucket is in a new _region_, not a new google project